types/epoch.go

```diff
@@ -48,11 +47,24 @@ func (e *Epoch) String() string {
 		return e.name
 	}
 
-	return hexutil.EncodeBig(e.number)
+	return e.number.String()
+}
+
+// IsEqual check if e equals input
+func (e *Epoch) IsEqual(input *Epoch) bool {
```

Comment:
In golang naming convention, `Equals` is a better name.

Also suggest to rename parameter `input` to `other`